app/handler/post.go

```diff
@@ -0,0 +1,266 @@
+package handler
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/dinever/golf"
+	"github.com/dingoblog/dingo/app/model"
+	"github.com/dingoblog/dingo/app/utils"
+)
+
+func registerPostHandlers(app *golf.Application, routes map[string]map[string]interface{}) {
+	adminChain := golf.NewChain(JWTAuthMiddleware)
+	app.Get("/api/posts", APIPostsHandler(0, 10))
+	routes["GET"]["posts_url"] = "/api/posts"
+
+	app.Get("/api/posts/:post_id", APIPostHandler)
+	routes["GET"]["post_url"] = "/api/posts/:post_id"
+
+	app.Get("/api/posts/slug/:slug", APIPostSlugHandler)
+	routes["GET"]["post_slug_url"] = "/api/posts/slug/:slug"
+
+	app.Get("/api/posts/:post_id/comments", APIPostCommentsHandler)
+	routes["GET"]["post_comments_url"] = "/api/posts/:post_id/comments"
+
+	app.Get("/api/posts/:post_id/author", APIPostAuthorHandler)
+	routes["GET"]["post_author_url"] = "/api/posts/:post_id/author"
+
+	app.Get("/api/posts/:post_id/excerpt", APIPostExcerptHandler)
+	routes["GET"]["post_excerpt_url"] = "/api/posts/:post_id/excerpt"
+
+	app.Get("/api/posts/:post_id/summary", APIPostSummaryHandler)
+	routes["GET"]["post_summary_url"] = "/api/posts/:post_id/summary"
+
+	app.Get("/api/posts/:post_id/tag_string", APIPostTagStringHandler)
+	routes["GET"]["post_tag_string_url"] = "/api/posts/:post_id/tag_string"
+
+	app.Get("/api/posts/:post_id/tags", APIPostTagsHandler)
+	routes["GET"]["post_tags_url"] = "/api/posts/:post_id/tags"
+
+	app.Put("/api/posts", adminChain.Final(APIPostSaveHandler))
+	routes["PUT"]["post_save_url"] = "/api/posts"
+
+	app.Post("/api/posts/:post_id/publish", adminChain.Final(APIPostPublishHandler))
+	routes["POST"]["post_publish_url"] = "/api/posts/:post_id/publish"
+
+	app.Delete("/api/posts/:post_id", adminChain.Final(APIPostDeleteHandler))
+	routes["DELETE"]["post_delete_url"] = "/api/posts/:post_id"
+}
+
+func getPostFromContext(ctx *golf.Context, param ...string) (post *model.Post) {
+	post = new(model.Post)
+	if len(param) == 0 {
+		for _, p := range []string{"post_id", "slug"} {
```

Comment:
Won't this range statement only ever end up calling `getPostFromContext(ctx, "post_id")`, since the `post_id` is the first element in the slice and will always match the `post_id` in the switch statement down below?